Fix random failure due to duplicate unaligned root edges

wenyuzhao opened the pull request:
This PR fixes https://github.com/mmtk/mmtk-openjdk/issues/11 and https://github.com/mmtk/mmtk-openjdk/issues/12 by reducing duplicate `CodeBlob` root edges.

Fix #11 
Fix #12

wenyuzhao commented:
After this PR is merged, we need to update `minheap` numbers in running scripts, for both personal evaluations and CI.

This fix can make OpenJDK run on a much smaller heap without crashing.  For example, I can now run `xalan` with 50M heap instead of 150M.

qinsoon reviewed (commented):
`CodeBlobToOopClosure` was used in 6 places in the binding, and this PR removed 2 uses out of the 6. Can you explain why this would fix the issue of duplicated root edges? (Sorry I dont know much about the binding).

wenyuzhao commented:
> `CodeBlobToOopClosure` was used in 6 places in the binding, and this PR removed 2 uses out of the 6. Can you explain why this would fix the issue of duplicated root edges? (Sorry I dont know much about the binding).

You are right. I removed one more `CodeBlobToOopClosure` in `MMTkHeap::scan_thread_roots`. This method is used when enabling single-threaded root scanning.

So far I removed 3/6 of `CodeBlobToOopClosure`s. Among the remaining 3 uses,
* `MMTkHeap::scan_global_roots` and `MMTkHeap::scan_roots` are dead code and should be removed when fixing #51 .
* `CodeBlobToOopClosure` in `MMTkHeap::scan_code_cache_roots` is necessary to report all the code roots. Code roots will only be reported once in this method.
